backend/src/schema/resolvers/shout.spec.js

```diff
@@ -19,141 +19,148 @@ const mutationUnshoutPost = gql`
+          data: { shout: false },
+        })
+        await expect(query({ query: queryPost, variables })).resolves.toMatchObject({
+          data: { Post: [{ id: 'current-user-post-id', shoutedBy: [] }] },
+          errors: undefined,
+        })
+      })
     })
   })
-
-  describe('unshout foreign post', () => {
-    describe('unauthenticated shout', () => {
+  describe('unshout', () => {
+    describe('unauthenticated', () => {
       it('throws authorization error', async () => {
-        // shout
-        await clientUser1.request(mutationShoutPost, { id: 'p12345' })
-        // unshout
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationUnshoutPost, { id: 'p12345' })).rejects.toThrow(
-          'Not Authorised',
-        )
+        authenticatedUser = undefined
+        variables = { id: 'post-to-shout-id' }
+        await expect(mutate({ mutation: mutationUnshoutPost, variables })).resolves.toMatchObject({
+          errors: [{ message: 'Not Authorised!' }],
+        })
       })
     })
 
-    it('I unshout a post of another user', async () => {
-      // shout
-      await clientUser1.request(mutationShoutPost, { id: 'p12345' })
-      const expected = {
-        unshout: true,
-      }
-      // unshout
-      const res = await clientUser1.request(mutationUnshoutPost, { id: 'p12345' })
-      expect(res).toMatchObject(expected)
+    describe('authenticated', () => {
+      beforeEach(async () => {
+        currentUser = await factory.create('User', {
+          id: 'current-user-id',
+          name: 'Current User',
+          email: 'current.user@example.org',
+          password: '1234',
+        })
 
-      const { Post } = await clientUser1.request(gql`
-        query {
-          Post(id: "p12345") {
-            shoutedByCurrentUser
-          }
-        }
-      `)
-      const expected2 = {
-        shoutedByCurrentUser: false,
-      }
-      expect(Post[0]).toMatchObject(expected2)
+        postAuthor = await factory.create('User', {
+          id: 'id-of-another-user',
+          name: 'Another User',
+          email: 'another.user@example.org',
+          password: '1234',
+        })
+        authenticatedUser = await currentUser.toJson()
+        await factory.create('Post', {
+          name: 'Posted By Another User',
+          id: 'posted-by-another-user',
+          author: postAuthor,
+        })
+        await mutate({
+          mutation: mutationShoutPost,
+          variables: { id: 'posted-by-another-user' },
+        })
+        variables = {}
+      })
+
+      it("another user's post", async () => {
+        variables = { id: 'posted-by-another-user' }
+        await expect(mutate({ mutation: mutationUnshoutPost, variables })).resolves.toMatchObject({
+          data: { unshout: true },
+        })
+        await expect(query({ query: queryPost, variables })).resolves.toMatchObject({
```

Comment:
You can avoid false negatives by repeating the assertion *before* and *after* you do the action. Consider that it might be `shoutedBy: []` already before you called the mutation.

In ruby land we would have the `expect{ ... }.to change{ ... }` matcher. In JavaScript, I'm afraid, you have to repeat a method before and after you do sth.